### PR TITLE
Contribution counter

### PR DESCRIPTION
**Pull request by sabinevidal**
Github workflow which checkouts the Kindly repo and calculates the number of phrases in the training_data.json

### CODE REVIEW

**Comment by sabinevidal** [comment]
@nathanfletcher: What would the best approach be to update the `counter` variable on the contributions page? 
I was thinking of writing a function which changes the variable, that the workflow runs as a script.

**Comment by nathanfletcher** [comment]
Yeah this sounds great.
If you intend to let  all this runs as a github workflow periodically (say daily) that would be best in my opinion.

**Comment by sabinevidal** [comment]
@lacabra @nathanfletcher: 
Workflow now rewrites the integer in the `COUNTER` variable in the `ContributionsCounter.js`. 
Have created a component which I've styled simply. 

Will change the workflow to run once a day an hour after the training data update workflow runs. 

What is the best way to commit the updated changes in `ContributionsCounter.js`? I found an action which does this, but not sure if this is the best way.

**Comment by sabinevidal** [comment]
@lacabra @nathanfletcher : 
Have now added `git commit` to the workflow using my password and username for now. 
And have updated the schedule to run the cron 15min after the `training_data` file is updated but eh workflow in the Kindly repo.

Ready for review!

**Comment by lacabra** [comment]
@sabinevidal can you confirm that you have run this script locally (without GitHub Actions) and it runs as expected?

**Comment by sabinevidal** [comment]
> @sabinevidal can you confirm that you have run this script locally (without GitHub Actions) and it runs as expected?

Just needed to correct a spacing issue that changed when formatting! Works as expected

**Comment by nathanfletcher** [comment]
@lacabra @sabinevidal for this PR and [kindly 114](https://github.com/nektos/act) I'm using [act](https://github.com/nektos/act) for testing locally. We'll sync on it @sabinevidal

File: .github/workflows/contribution-counter.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - name: Get current date
+        id: date
+        run: echo "::set-output name=today::$(date +'%Y-%m-%d')"
+
+      - name: 'Commit updated counter'
+        run: |
+          git config --global user.name 'sabinevidal'
+          git config --global user.email 'sabinevidal@users.noreply.github.com'
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/unicef/kindly-website
+          git commit -am "Updated counter - ${{ steps.date.outputs.today }}"
```
Comment (nathanfletcher) [inline .github/workflows/contribution-counter.yml:44]: update commit message to include count

**Comment by nathanfletcher** [comment]
@lacabra We will be testing this in conjunction with [kindly PR 114](https://github.com/unicef/kindly/pull/114) on different cloned repos before merging.

File: .github/workflows/contribution-counter.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Contribution Counter
+
+on:
+  schedule:
```
Comment (lacabra) [inline .github/workflows/contribution-counter.yml:4]: Why is this line `schedule` not commented out ? when does this action run if it has a `schedule` section with no data in it?

Comment (sabinevidal) [reply .github/workflows/contribution-counter.yml:4]: Have uncommented the cron now. Had commented it out during my testing until all looked good.

**Comment by sabinevidal** [comment]
> > @lacabra @nathanfletcher : Have now added `git commit` to the workflow using my password and username for now. And have updated the schedule to run the cron 15min after the `training_data` file is updated but eh workflow in the Kindly repo.
> 
> @sabinevidal: can you point to the specific lines in the code that trigger the workflow "to run the cron 15min after `training_data` file is updated"? This piece is not obvious to me. Thanks

The cron is set to run at 00:15 UTC which is 15min after the Update Dataset workflow on the Kindly repo to capture the changes.

File: .github/workflows/contribution-counter.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    # Runs once a day at 00:15 UTC
+    - cron: '15 0 * * *'
```
[the remark this answers is not in the document]
Comment (lacabra) [reply .github/workflows/contribution-counter.yml:6]: I see, thank you.

**Review by lacabra — APPROVED** [review]
This looks good to me. Thank you @sabinevidal 🙏

**Pull request merged**